### PR TITLE
feat: added structure of catalog

### PR DESCRIPTION
**Pull request by SolMendiola**
# :star: Feature

---

#### :pencil2: Description:

Add catalog package #186 
 
---

### CODE REVIEW

File: catalog/gallery/lib/catalog/catalog_app_button_screen.dart [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import 'package:auto_route/auto_route.dart';
 import 'package:flutter/material.dart';
-import 'package:gallery/catalog/catalog_scaffold_screen.dart';
+import 'catalog_scaffold_screen.dart';
```
Comment (mirland) [inline catalog/gallery/lib/catalog/catalog_app_button_screen.dart]: You should use full, maybe your rules are not ok

File: catalog/gallery/lib/catalog/catalog_scaffold_screen.dart [hunk 2]
```diff
@@ -0,0 +1,37 @@
+import 'package:auto_route/auto_route.dart';
+import 'package:flutter/material.dart';
+import 'package:catalog/catalog.dart';
+
+@RoutePage()
+class CatalogScaffoldScreen extends StatelessWidget {
```
Comment (nicolantean) [inline catalog/gallery/lib/catalog/catalog_scaffold_screen.dart:6]: I think it's jus CatalogScaffold without screen

File: catalog/gallery/lib/main/catalog_main_screen.dart [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+import 'package:catalog/catalog.dart';
+import 'package:flutter/material.dart';
+import 'package:gallery/router/catalog_router.dart';
+
+@RoutePage()
+class CatalogMainScreen extends StatelessWidget {
+  const CatalogMainScreen({super.key});
+
+  @override
+  Widget build(BuildContext context) => Scaffold(
```
Comment (nicolantean) [inline catalog/gallery/lib/main/catalog_main_screen.dart:11]: maybe you can use the CatalogScaffold

File: catalog/gallery/pubspec.yaml [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+# To automatically upgrade your package dependencies to the latest versions
+# consider running `flutter pub upgrade --major-versions`. Alternatively,
+# dependencies can be manually updated by changing the version numbers below to
+# the latest version available on pub.dev. To see which dependencies have newer
+# versions available, run `flutter pub outdated`.
+dependencies:
+  flutter:
+    sdk: flutter
+  catalog:
+    path: ../../catalog
```
Comment (nicolantean) [inline catalog/gallery/pubspec.yaml:20]: this is the same as doing ../ right?

**Review by nicolantean — APPROVED** [review]

File: catalog/gallery/lib/main/catalog_section.dart [hunk 5]
```diff
@@ -0,0 +1,14 @@
+import 'package:auto_route/auto_route.dart';
+import 'package:flutter/widgets.dart';
+
+@RoutePage(name: 'CatalogSectionRouter')
```
Comment (mirland) [inline catalog/gallery/lib/main/catalog_section.dart:4]: Do you need it 🤔

File: catalog/gallery/lib/router/catalog_router.dart [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+  @override
+  final List<AutoRoute> routes;
+
+  CatalogRouter()
+      : routes = [
+          AutoRoute(
+            page: CatalogSectionRouter.page,
+            path: '/',
+            children: [
+              RedirectRoute(path: '', redirectTo: 'main'),
```
Comment (mirland) [inline catalog/gallery/lib/router/catalog_router.dart:22]: use initial

File: catalog/lib/theme/app_theme.dart [hunk 7]
```diff
@@ -0,0 +1,32 @@
+import 'package:catalog/theme/app_text_styles.dart';
+import 'package:flutter/material.dart';
+import 'package:catalog/theme/app_colors.dart';
+import 'package:catalog/theme/app_dimensions.dart';
+
+late AppColors _colors;
+late AppDimens _dimensions;
+late AppTextStyles _styles;
```
Comment (mirland) [inline catalog/lib/theme/app_theme.dart:8]: Add task to use theme extensions

File: catalog/pubspec.yaml [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+dev_dependencies:
+  flutter_test:
+    sdk: flutter
+  flutter_lints: 3.0.1
+  auto_route_generator: 7.3.1
+  flutter_gen_runner: 5.3.1
+  build_runner: 2.4.6
+  lints: 3.0.0
+
+flutter:
```
Comment (mirland) [inline catalog/pubspec.yaml:27]: do you need it?

File: lib/ui/theme/text_styles.dart [hunk 9]
```diff
@@ -7,30 +7,19 @@ class AppStyles extends TextTheme {
 
   const AppStyles({
```
Comment (mirland) [inline lib/ui/theme/text_styles.dart:8]: why? You should use the catalog theme

**Review by mirland — APPROVED** [review]

**Pull request merged**